src/v/cloud_storage/cache_service.cc

```diff
@@ -1288,42 +1289,55 @@ ss::future<> cache::put(
     options.io_priority_class = io_priority;
     auto out = co_await ss::make_file_output_stream(tmp_cache_file, options);
 
-    std::exception_ptr disk_full_error;
+    std::exception_ptr eptr;
+    bool no_space_on_device = false;
     try {
         co_await ss::copy(data, out)
           .then([&out]() { return out.flush(); })
           .finally([&out]() { return out.close(); });
     } catch (const std::filesystem::filesystem_error& e) {
         // For ENOSPC errors, delay handling so that we can do a trim
-        if (e.code() == std::errc::no_space_on_device) {
-            disk_full_error = std::current_exception();
-        } else {
-            throw;
-        }
+        no_space_on_device = e.code() == std::errc::no_space_on_device;
```

Comment:
nit: IMO the non-test changes in this PR are small/tightly scoped to all have been in the same commit

Reply:
Got it, I appreciate that. My personal line for when/where to split is just different -- I typically separate like that when the PR is more complex and the cosmetic changes get in the way of review. I didn't really feel that was the case here, and I actually reviewed the PR as a whole (non-test changes here are like ~40 lines of straightforward code)

No need to change, it's subjective ultimately